src/classes/class-se-calendar.php

```
@@ -340,57 +369,35 @@ private function get_events_by_date( $date ): array {
-			$start_timestamp
-		);
-
-		// phpcs:disable WordPress.DB.PreparedSQL.NotPrepared
-		$all_events = $wpdb->get_results( $sql_query ); // the query is prepared above
-		if ( $all_events ) {
-			foreach ( $all_events as $event ) {
-				$event_dates = se_event_get_dates( $event->ID );
+		$new_all = SE_Event_Dates::get_event_dates_for_date( $date->format( 'Y-m-d' ), true, false );
+		// If we new dates.
+		if ( ! empty( $new_all ) ) {
+			foreach ( $new_all as $index => $event ) {
```

PHPMD (2.15.0)

Avoid unused local variables such as '$index'. (Unused Code Rules)

(UnusedLocalVariable)